src/utils/updateTokenPrices.ts

```diff
@@ -103,3 +115,104 @@ async function getVeth2Price(
     return etherPrice
   }
 }
+
+export const getTokenPrice = (
+  tokens: BasicTokens,
+  dispatch: AppDispatch,
+  chainId: ChainId,
+): void => {
+  if (!tokens) return
+  const tokenAddresses = Object.keys(tokens)
+  const platform =
+    SUPPORTED_NETWORKS[chainId]?.coingeckoPlatformID || "ethereum"
+  const addressesChunk = chunk(
+    tokenAddresses,
+    MAX_ADDRESSES_PER_COINGECKO_REQUEST,
+  )
+  if (SUPPORTED_NETWORKS[chainId]) {
+    try {
+      void retry(
+        async () => {
+          const pricesChunks = await Promise.all(
+            addressesChunk.map((chunkedAddress) =>
+              fetch(
+                `https://api.coingecko.com/api/v3/simple/token_price/${platform}?contract_addresses=${chunkedAddress.join(
+                  ",",
+                )}&vs_currencies=usd`,
+              )
+                .then((res) => res.json())
+                .then((prices: CoinGeckoReponse) =>
+                  arrayToHashmap(
+                    Array.from(
+                      Object.entries(prices).map(
+                        ([address, { usd: usdPrice }]) => [
+                          address.toLocaleLowerCase(),
+                          usdPrice,
+                        ],
+                      ),
+                    ),
+                  ),
+                ),
+            ),
+          )
+
+          const tokenPricesUSD = Object.assign(
+            {},
+            ...pricesChunks,
+          ) as TokenPricesUSD
+          dispatch(updateTokensPricesUSD(tokenPricesUSD))
+        },
+        { retries: 3 },
+      )
+    } catch (error) {
+      console.error("Error on fetching price from coingecko ==>", error)
+    }
+  } else {
+    try {
+      void retry(
+        () =>
+          fetch(`${coinGeckoAPI}?ids=${encodeURIComponent(
+            nativeTokenID.join(","),
+          )}&vs_currencies=usd
+    `)
+            .then((res) => res.json())
+            .then((data: CoinGeckoReponse) => {
+              const nativeTokenPrice = {
+                btc: data.bitcoin.usd,
+                eth: data.ethereum.usd,
+                usd: data.tether.usd,
+              }
+              const tokenPricesUSD = Object.assign(
+                {},
+                ...Object.keys(tokens).map((tokenAddress) => {
+                  switch (tokens[tokenAddress]?.typeAsset) {
+                    case PoolTypes.BTC:
+                      return { [tokenAddress]: nativeTokenPrice.btc }
+                      break
+
+                    case PoolTypes.ETH:
+                      return { [tokenAddress]: nativeTokenPrice.eth }
+                      break
+                    case PoolTypes.USD:
+                      return { [tokenAddress]: nativeTokenPrice.usd }
+                      break
+                    default:
```

Comment:
for localhost I'd like us to use a placeholder value for `OTHER` or unknown tokens. Any number you want.

Reply:
Do you want me to give any value for `OTHER` or `unknown` token price?